### ISSUE TITLE
Venia splash and loading experience (MVP)

### ISSUE THREAD

**Issue by ericerway** [magento-research/pwa-studio#461]
## This issue is for the following packages:

[X] `venia-concept`
[ ] `pwa-buildpack`
[ ] `peregrine`
[ ] `pwa-devdocs`
[ ] `upward-js`
[ ] `upward-spec`

## This issue is a:

[ ] Bug
[ ] Feature suggestion
[ ] Documentation issue
[ ] Other (Please Specify)

### Environment

| Question                    |    Answer     |
| --------------------------- | :-----------: |
| Magento version             |  |
| Operating System + version  |  |
| node.js version (`node -v`) |  |
| npm version (`npm -v`)      |  |

## Description

(builds upon user experience work in #468)

As a shopper, I want to have a subtle experience on loading when I arrive at the Venia storefront (or new page) from a browser or app launched. This is a requirement for the Google Lighthouse PWA definition.

Two scenarios should be considered:

- First load of Venia from app homescreen/doc (as a chromeless app or PWA)
- First load of Venia pages (Product Details, etc.)

This can be one or two different experiences to address these scenarios. Additional information can be found on Google here: https://developers.google.com/web/fundamentals/web-app-manifest/#splash-screen

## Expected result:

No messaging about loading or fetching data... to be replaced with a more subtle indication (or splash) as appropriate.

## Possible solutions:

TBD

**Comment by dmtrkad** [comment]
@ericerway , @awilcoxa , Could you please update this ticket with referebce to desgign we need to follow? thx

**Comment by ericerway** [comment]
Absolutely. We need a quick discussion this week to inform what this looks like with @soumya-ashok and team. Thanks!

**Comment by ericerway** [comment]
Simplified and assigning @jimbo to investigate. Let's look at a basic first iteration and revisit shortly afterwards. Thanks!

**Comment by Starotitorov** [comment]
@ericerway In my opinion, we can split this story into smaller ones.
As far as I understand, we have three scenarios and it is possible to have separate designs for them.
1. Splash screen. We need to show it when user loads Venia app from homescreen.
2. Loading indicator when we execute GraphQL query `urlResolver` to understand what page we need to render for the url. It is a loading indicator for the whole page.
3. Indicator to show progress loading some data on the page. For example, when we render a collection of elements and the request to fetch items is in progress.

@ericerway, @awilcoxa, @soumya-ashok We need a designs for all this cases.

**Comment by Starotitorov** [comment]
@ericerway, @awilcoxa, @soumya-ashok, splash screen have already been added, all required information has already been added to manifest.json:

![splash-screen](https://user-images.githubusercontent.com/11873143/49521954-8b9fa800-f8b7-11e8-953b-982d2cf1e328.png)

**Comment by Starotitorov** [comment]
So, we found out that splash screen has already been implemented. @ericerway, @soumya-ashok. we need designs to start implementing:

1. Loading indicator when we execute GraphQL query urlResolver to understand what page we need to render for the url. It is a loading indicator for the whole page.
2. Indicator to show progress loading some data on the page. For example, when we render a collection of elements and the request to fetch items is in progress.

@ericerway, right now we are blocked.
